docs/source/site-specific-config.rst

```diff
@@ -0,0 +1,142 @@
+.. _site_specific_config:
+
+Site-Specific Configuration
+***************************
+A site might have compilers that Fab doesn't know about, or prefers
+a different compiler from the Fab default. Fab abstracts the compilers
+and other programs required during building as an instance of a
+:class:`~fab.tools.tool.Tool` class. All tools that Fab knows about, are
+available in a :class:`~fab.tools.tool_repository.ToolRepository`.
+That will include tools that might not be available on the current system.
+
+Each tool belongs to a certain category of
+:class:`~fab.tool.categories.Categories`. A `ToolRepository` can store
+several instances of the same category.
+
+At build time, the user has to create an instance of
+:class:`~fab.tools.tool_box.ToolBox` and pass
+it to the :class:`~fab.build_config.BuildConfig` object. This toolbox
+contains all the tools that will be used during the build process, but
+it can only store one tool per category. If a certain tool should not
```

Comment:
There are two approaches:
1. (which is what we use) We create `Meta` compiler class: its constructor takes two existing compilers instances (e.g. normal compiler and profile-wrapper), and its compile method will decide which compiler to use (e.g. based on path, which is what we use, we exclude all `.../kernels/...` from being instrumented, since they are called way too frequent). This could even be based on a static/dynamic dictionary being defined (similar to path flags)
2. The user could call the compile step more than once (with different source getter), and modify the toolbox in between. The problem with this is that it would not take dependencies into account, i.e. can all files be compiled with one compiler before the other??

I think the use case of this is reasonable infrequent that the first approach is good enough (esp. since you could write a constructor for a meta compiler that specifies which compiler to use for which path)

Reply:
My wording was perhaps a bit bad ... meta-compiler does not clearly describe what I meant :)   I totally agree, wrapped compilers are VERY important. I was meant to focus on 'specific compiler for specific files' (e.g. what we need for profiling, where we don't want to instrument kernels) are being rare. Seeing that we agree in #312, I am closing this thread then.